dali/pipeline/util/bounding_box.h

```diff
@@ -153,6 +156,25 @@ class BoundingBox {
     return {left_, top_, right_ - left_, bottom_ - top_};
   }
 
+  void CopyAsCenterWh(float *destination) const {
+    destination[0] = (left_ + right_) * 0.5f;
+    destination[1] = (top_ + bottom_) * 0.5f;
+    destination[2] = right_ - left_;
+    destination[3] = bottom_ - top_;
+  }
+
+  static vector<BoundingBox> FromLtrbArray(const float *source, int num, bool check_input = true) {
```

Comment:
I'm not sure if we need this function at all. You can use FromLtrb(float*) as it is from the outside of this class. If you see fit for a function like this (which might, as it might become tiring to copypaste the for loop to parse a buffer), you could have it as a free standing functionin a namespace BoundingBox (BoundingBox::FromLtrbBuffer(float* ptr, size_t box_count))

The concept of these static functions is to deter from using directly the constructor. It is not very consistent with this premise to have a static function that calls another static function that calls the constructor in order to get a container of objects. Since what you are trying to do is to wrap the initialization of a container and the multiple calls to FromLtrb, I think a detached function would be the best

Reply:
Fixed by moving this functionality as `ReadBoxesFromInput` method of `BoxEncoder`.